engine/install.go

```diff
@@ -126,6 +128,20 @@ poller:
 		Str("image", i.image).
 		Msg("create agent container")
 
+	volumes := map[string]struct{}{
+		"/var/run/docker.sock": {},
+	}
+
+	binds := []string{
+		"/var/run/docker.sock:/var/run/docker.sock",
+	}
+
+	for _, volume := range i.volumes {
+		splitted := strings.Split(volume, ":")
+		volumes[splitted[1]] = struct{}{}
+		binds = append(binds, volume)
+	}
```

Comment:
lets re-use this helper function:
```
// helper function that converts a slice of volume paths to a set of
// unique volume names.
func toVol(paths []string) map[string]struct{} {
	set := map[string]struct{}{}
	for _, path := range paths {
		parts, err := splitVolumeParts(path)
		if err != nil {
			continue
		}
		if len(parts) < 2 {
			continue
		}
		set[parts[1]] = struct{}{}
	}
	return set
}
```

and also lets import the unit test:
https://github.com/cncd/pipeline/blob/master/pipeline/backend/docker/convert_test.go#L78

Reply:
@bradrydzewski 
done - however I am a bit concerned with the authorship - code is partially from you and partially from another contributor